CONFIG.md
# Environment Variables

## Version

Version info

 - `SOURCE_BRANCH` - 
 - `SOURCE_COMMIT` - 
 - `IMAGE_NAME` - 

## Config

Application configuration

 - `MODE` (default: `production`) - Env must be local, development, test or production
 - `BASE_URL` - 
 - `TIMEZONE` (default: `Europe/Brussels`) - 
 - `INDEX_RETENTION` (default: `2`) - 
 - `PG_CONN` - 
 - `ES6_URL` - 
 - `PUBLICATION_INDEX` - 
 - `DATASET_INDEX` - 
 - `HOST` - 
 - `PORT` (default: `3000`) - 
 - 
   - `API_HOST` - 
   - `API_PORT` (default: `30000`) - 
 - `ADMIN_USERNAME` - 
 - `ADMIN_PASSWORD` - 
 - `CURATOR_USERNAME` - 
 - `CURATOR_PASSWORD` - 
 - `IP_RANGES` - 
 - 
   - `S3_ENDPOINT` - 
   - `S3_REGION` (default: `us-east-1`) - 
   - `S3_ID` - 
   - `S3_SECRET` - 
   - `S3_BUCKET` - 
   - `S3_TEMP_BUCKET` - 
 - 
   - `SESSION_NAME` (default: `biblio-backoffice`) - 
   - `SESSION_SECRET` - 
   - `SESSION_MAX_AGE` (default: `2592000`) - default 30 days
 - 
   - `CSRF_NAME` (default: `biblio-backoffice.csrf-token`) - 
   - `CSRF_SECRET` - 
 - `MAX_FILE_SIZE` (default: `2000000000`) - 
 - `FILE_DIR` - 
 - 
   - `FRONTEND_URL` - 
   - `FRONTEND_USERNAME` - 
   - `FRONTEND_PASSWORD` - 
   - `FRONTEND_ES6_URL` - 
 - 
   - `ORCID_CLIENT_ID` - 
   - `ORCID_CLIENT_SECRET` - 
   - `ORCID_SANDBOX` - 
 - 
   - `OIDC_URL` - 
   - `OIDC_CLIENT_ID` - 
   - `OIDC_CLIENT_SECRET` - 
 - `CITEPROC_URL` - 
 - `MONGODB_URL` - 
 - 
   - `HDL_SRV_ENABLED` - 
   - `HDL_SRV_URL` - 
   - `HDL_SRV_PREFIX` - 
   - `HDL_SRV_USERNAME` - 
   - `HDL_SRV_PASSWORD` - 
   - `HDL_SRV_API_URL` - 
   - `HDL_SRV_API_KEY` - 
 - 
   - `OAI_ENABLED` - 
   - `OAI_URL` - 
   - `OAI_PREFIX` - 
   - `OAI_USERNAME` - 
   - `OAI_PASSWORD` - 
   - `OAI_API_URL` - 
   - `OAI_API_KEY` - 
 - 
   - `PROJECTS_API_URL` - 
   - `PROJECTS_API_KEY` - 
 - 
   - `PEOPLE_API_URL` - 
   - `PEOPLE_API_KEY` - 
 - Feature flags
   - `FF_FILE_PATH` - 
   - `FF_GITHUB_TOKEN` - 
   - `FF_GITHUB_REPO` - 
   - `FF_GITHUB_BRANCH` (default: `main`) - 
   - `FF_GITHUB_PATH` - 


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added

### Fixed

## [v1.0.52] - 2024-01-31

### Added

- #1360: Add EGA identifier for manual import of datasets

### Fixed

## [v1.0.51] - 2024-01-22

### Added

### Fixed

- Fix bug in cleanup command

## [v1.0.50] - 2024-01-22

### Added

### Fixed

- Improve OAI export
- Fix bug in cleanup command

## [v1.0.49] - 2024-01-19

### Added

- #1354: add CSP headers

### Fixed

## [v1.0.48] - 2023-12-18

### Added

### Fixed

- #1343: Switch to a more lenient BibTeX parser
- #1345: Fix dataset redirect url

## [v1.0.47] - 2023-11-29

### Added

### Fixed

- #1332: Path escape project id's when necessary

## [v1.0.46] - 2023-11-27

### Added

### Fixed

- #1330: Fix setting HasBeenPublic for publications

## [v1.0.45] - 2023-11-23

### Added

### Fixed

- Fix display of missing dissertation translation strings

## [v1.0.44] - 2023-11-22


### Added

### Fixed

- Fix display of missing dataset translation strings

## [v1.0.43] - 2023-11-22

### Added

### Fixed

- Fix display of missing publication translation strings

## [v1.0.42] - 2023-11-22

### Added

- User dashboard
- More Cypress tests
- Various application structure improvements

### Fixed

- Various ux improvements
- #1223: Cleanup empty facet values
- #1241: Only allow future embargo dates

## [v1.0.41] - 2023-11-15

### Added

- Add status.set, lock, unlock batch operations
- Determine subtype based on wos type
- Move frontoffice index-time enrichments to backoffice record

### Fixed

- Fix sidebar color switching
- #1299: Handle out of range pagination

## [v1.0.40] - 2023-10-17

### Added

### Fixed

- #1267: Fix edit roles dialog

## [v1.0.39] - 2023-10-17

### Added

- Cypress testing improvements
- Switch to Bootstrap 5
- Application structure simplifications and improvements
- Various ux improvements
- #1237: Various accessibility improvements
- #1227: Make conference name searchable

### Fixed

- #1223: Fix empty librarian tag facet values

## [v1.0.38] - 2023-09-22

### Added

### Fixed

- Fix regression in searchable fields

## [v1.0.37] - 2023-09-20

### Added

- Start bringing code structure in line with our other apps (deliver, oai-service and people-service)

### Fixed

- #1227: Fix regression in searchable fields

## [v1.0.36] - 2023-09-19

### Added

### Fixed

- #1226: Fix lock record button after edit

## [v1.0.35] - 2023-09-19

### Added

- #1110: Keep related object metadata up-to-date
- #812: remove confirmation dialog for Lock and Unlock
- Smaller and more efficient search index
- Add vabb_id.set, vabb_type.set, vabb_approved.set batch operations
- Various ux improvements
- remove downtime banner

### Fixed

- #1108: Cannot add author without first name

## [v1.0.34] - 2023-09-15

### Added

- add planned downtime banner

### Fixed

## [v1.0.33] - 2023-06-30

### Added

### Fixed

- Lower limit for suggest to 20

## [v1.0.32] - 2023-06-20

### Added

- Only save a new version of a publication or dataset if the record has changes

### Fixed

## [v1.0.31] - 2023-06-20

### Added

- Only save a new version of a publication or dataset if the record has changes

### Fixed

## [v1.0.30] - 2023-06-20

### Added

- Add vabb_id.set, vabb_type.set, vabb_approved.set batch operations

### Fixed

## [v1.0.29] - 2023-06-13

### Added

### Fixed

- Allow HTML in form help text
- Update text

## [v1.0.28] - 2023-06-13

### Added

- #898: Manual dataset input and support dataset identifier types other than DOI
- Batch operation support (initial operations are project.add,
  classification.set, vabb_year.add, keyword.add, keyword.remove,
  reviewer_tag.add and reviewer_tag.remove)
- #1109: Map VABB attributes in frontoffice handler

### Fixed

- #1123: Fix import of multiline RIS/WoS fields
- Various ux fixes

## [v1.0.27] - 2023-05-08

### Added

### Fixed

- Small ux fix

## [v1.0.26] - 2023-05-08

### Added

- S3 support for file storage
- #1034: Add sync-publication-contributors command to keep embedded personal
  details up to date.
- #973: Add link to WoS ID on record detail page
- #1081: support HEAD request for file downloads
- #1065: Add FlandersMake organization
- #1058: Hide deprecated DefenseTime field
- #601: Add unclassified and unaffiliated records to dashboard
- More Cypress tests

### Fixed

- #1083: Increase request timeout to 5 minutes from 3 minutes
- #1051: WoS import does not import all multiline fields completely
- Ensure contact details are always visible
- Various small ux fixes

## [v1.0.25] - 2023-03-30

### Added

- gRPC client is now in a usable state

### Fixed

- #1039: Make interaction with authority database more robust
- Various small ux fixes

## [v1.0.24] - 2023-03-21

### Added

### Fixed

- #1035: DOI import fails, leaves blank screen for 2 DOIs
- #1030: Cannot see publications linked to a dataset as a researcher
- #1028: Fix editing of keyword tag

## [v1.0.23] - 2023-03-08

### Added

### Fixed

- Always show VABB export years

## [v1.0.22] - 2023-02-17

### Added

- #1017: Add locked / unlocked filter to datasets
- #983: File upload progress
- Show error dialog if upload is too large
- Make maximum file size configurable
- #930: Reindex command
- #955: Better WOS Type facet (reindex needed)
- #1015: Search on organization ID (reindex needed)
- #956: Show status of facet filter in badge
- #1004: Add Reset filters button
- #957: Show top 3 selected facet values in badge
- Cypress tests

### Fixed

- Fix conflict error after file upload cancel
- #1006: Expose abstract language to frontoffice

## [v1.0.21] - 2023-01-25

### Added

### Fixed

- Fix bug in some repeated text fields

## [v1.0.20] - 2023-01-25

### Added

- Switch from deprecated biblio-backend prefix to biblio-backoffice

### Fixed


cli/config.go
package cli

// Version info
type Version struct {
	Branch string `env:"SOURCE_BRANCH"`
	Commit string `env:"SOURCE_COMMIT"`
	Image  string `env:"IMAGE_NAME"`
}

// Application config
type Config struct {
	// Env must be local, development, test or production
	// TODO rename MODE to ENV
	Env              string `env:"MODE" envDefault:"production"`
	BaseURL          string `env:"BASE_URL"`
	Timezone         string `env:"TIMEZONE" envDefault:"Europe/Brussels"`
	IndexRetention   int    `env:"INDEX_RETENTION" envDefault:"2"`
	PgConn           string `env:"PG_CONN"`
	Es6URL           string `env:"ES6_URL"`
	PublicationIndex string `env:"PUBLICATION_INDEX"`
	DatasetIndex     string `env:"DATASET_INDEX"`
	Host             string `env:"HOST"`
	Port             int    `env:"PORT" envDefault:"3000"`
	API              struct {
		Host string `env:"HOST"`
		Port int    `env:"PORT" envDefault:"30000"`
	} `envPrefix:"API_"`
	AdminUsername   string `env:"ADMIN_USERNAME"`
	AdminPassword   string `env:"ADMIN_PASSWORD"`
	CuratorUsername string `env:"CURATOR_USERNAME"`
	CuratorPassword string `env:"CURATOR_PASSWORD"`
	IPRanges        string `env:"IP_RANGES"`
	S3              struct {
		Endpoint   string `env:"ENDPOINT"`
		Region     string `env:"REGION" envDefault:"us-east-1"`
		ID         string `env:"ID"`
		Secret     string `env:"SECRET"`
		Bucket     string `env:"BUCKET"`
		TempBucket string `env:"TEMP_BUCKET"`
	} `envPrefix:"S3_"`
	Session struct {
		Name   string `env:"NAME" envDefault:"biblio-backoffice"`
		Secret string `env:"SECRET"`
		MaxAge int    `env:"MAX_AGE" envDefault:"2592000"` // default 30 days
	} `envPrefix:"SESSION_"`
	CSRF struct {
		Name   string `env:"NAME" envDefault:"biblio-backoffice.csrf-token"`
		Secret string `env:"SECRET"`
	} `envPrefix:"CSRF_"`
	MaxFileSize int    `env:"MAX_FILE_SIZE" envDefault:"2000000000"`
	FileDir     string `env:"FILE_DIR"`
	Frontend    struct {
		URL      string `env:"URL"`
		Username string `env:"USERNAME"`
		Password string `env:"PASSWORD"`
		Es6URL   string `env:"ES6_URL"`
	} `envPrefix:"FRONTEND_"`
	ORCID struct {
		ClientID     string `env:"CLIENT_ID"`
		ClientSecret string `env:"CLIENT_SECRET"`
		Sandbox      bool   `env:"SANDBOX"`
	} `envPrefix:"ORCID_"`
	OIDC struct {
		URL          string `env:"URL"`
		ClientID     string `env:"CLIENT_ID"`
		ClientSecret string `env:"CLIENT_SECRET"`
	} `envPrefix:"OIDC_"`
	CiteprocURL string `env:"CITEPROC_URL"`
	MongoDBURL  string `env:"MONGODB_URL"`
	APIKey      string `env:"API_KEY"`
	Handle      struct {
		Enabled  bool   `env:"ENABLED"`
		URL      string `env:"URL"`
		Prefix   string `env:"PREFIX"`
		Username string `env:"USERNAME"`
		Password string `env:"PASSWORD"`
	} `envPrefix:"HDL_SRV_"`
	OAI struct {
		APIURL string `env:"API_URL"`
		APIKey string `env:"API_KEY"`
	} `envPrefix:"OAI_"`
	Projects struct {
		APIURL string `env:"API_URL"`
		APIKey string `env:"API_KEY"`
	} `envPrefix:"PROJECTS_"`
	People struct {
		APIURL string `env:"API_URL"`
		APIKey string `env:"API_KEY"`
	} `envPrefix:"PEOPLE_"`
	// Feature flags
	FF struct {
		FilePath     string `env:"FILE_PATH"`
		GitHubToken  string `env:"GITHUB_TOKEN"`
		GitHubRepo   string `env:"GITHUB_REPO"`
		GitHubBranch string `env:"GITHUB_BRANCH" envDefault:"main"`
		GitHubPath   string `env:"GITHUB_PATH"`
	} `envPrefix:"FF_"`
}
